### DISCUSSION TITLE
Маска для месяца и года (MM/YY) с указанием ввода допустимых значений

### DISCUSSION THREAD

**Discussion by photomakes** [beholdr/maska#191] [Q&A] [answered]
Пробую сделать маску, которая допускаем ввод только подходящих значений для месяца и года.

Стоит отметить, что речь идёт о поле срока действия карты.

Месяц — число от 1 до 12.
Год — последние две цифры года, не меньше текущего года.

Пробовав разные варианты и пытаясь понять, что именно мне подходит из документации для достижения цели, остановился на динамической маске.

Создал три токена и, определяя первое вводимое значение, подставляю подходящую маску.

1. Если ввод начинается с 0, то подставляется маска с 0 и паттерн для следующего символа 1-9.
2. Если ввод начинается с 1, то подставляется маска MM, почему-то маска 1M не работает, как для условия с 0 (0m) из пункта 1.
3. Если вовод начинается с числа больше 1, то подставляется маска 0m, подразумевается, что вводится месяц от 1 до 9 без цифры 0.
4. Условие, защищающее от ввода чего-либо кроме цифр.

После заполнения месяца, хочется и с годом работать, чтобы можно было ввести год не меньше текущего. Но всё, до чего смог пока додуматься, это лишь задание паттерна для первой цифры, а для второй дефолтный паттерн #.

Остаются два нерешённых момента.
1. При вводе месяца 1 (январь), ничего не поделаешь и нужно вводить именно 01.
2. Год можно ввести 20, вместо 23 (исходя из года, на момент написания этого текста).

Вот код:

	const Mask = Maska.Mask, MaskInput = Maska.MaskInput, vMaska = Maska.vMaska;
	new MaskInput("[data-maska]");

	new MaskInput(".card-date", {
		tokens: {
			m: { pattern: /[1-9]/ },
			M: { pattern: /[0-2]/ },
			y: { pattern: /[2-9]/ },
		},
		mask: function (value) {
			if (value.startsWith('0')) {
				return '0m/y#';
			} else if (value.startsWith('1')) {
				return 'MM/y#';
			} else if (value.charAt(0) > 1) {
				return '0m/y#';
			} else {
				return '##/##';
			}
		},
	});

Я не программист и не владею js, поэтому не судите за написанный код, пожалуйста.

**Comment by beholdr** [comment, marked as answer]
I think it's best to use hooks for processing a complex logic like dates comparing.
I've made [working example](https://stackblitz.com/edit/vue-maska-pnholo?file=src%2FApp.vue) for given conditions.

**Reply by photomakes** [reply]
Спасибо!
